web/packages/teleport/src/Locks/useGetTargetData.tsx

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { useEffect, useState, useMemo } from 'react';
+
+import useTeleport from 'teleport/useTeleport';
```

Comment:
mocking `useTeleport` is problematic due to its internal structure, we probably want to rearchitect this sometime in the future.